File: Etiquette/lib/Providers/KAS/Kip17/kip17_get_token_data.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:Etiquette/Models/serverset.dart';

Future<Map<String, dynamic>> kip17GetTokenData(String alias, String token_id) async {
  const url = "$SERVER_IP/kas/kip17/getTokenData";
  try {
    var res = await http.post(Uri.parse(url), body: {
      "alias": alias,
      "token_id": token_id
    });
    Map<String, dynamic> data = json.decode(res.body);
    return data;
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}

File: Backend/controller/kas/kip17/kip17GetTokenData.js
const kip17 = require('../../../service/kas/kip17');

const kip17GetTokenData = async function (req, res) {
    try {
        const tokenData = await kip17.kasKip17GetTokenData(req.body.alias, req.body.token_id);

        if (tokenData['statusCode'] == 200) {
            res.status(200);
            res.json({statusCode: 200, data: tokenData['data']});
        } else {
            const errorCode = tokenData['data']['code'];
            res.status(tokenData['statusCode']);

            switch (errorCode) {
                case 1104404:
                    res.json({
                        statusCode: tokenData['statusCode'],
                        msg: "조회할 토큰이 존재하지 않습니다."
                    });
                    break;
                default:
                    res.json({
                        statusCode: tokenData['statusCode'],
                        msg: "토큰을 조회할 수 없습니다. 고객센터에 문의해 주세>요."
                    });
            }
        }
    } catch (e) {
        console.error(e);
    }
}

module.exports = kip17GetTokenData;